client/templates/courses/tagged-courses.js

```diff
@@ -6,3 +6,12 @@ Template.taggedCourses.helpers({
         return Session.get('tag');
     }
 });
+
+Template.taggedCourses.onCreated(function(){
+  // Accessing the Iron.controller to invoke getParams method of Iron Router.
+  var controller = Iron.controller();
+  // Getting Params of the URL
+  var params = controller.getParams();
+  // Invoking subscription on the Courses with the :tag params
+  Meteor.subscribe('taggedCourses', params.tag)
```

Comment:
Check the logic inside of the `taggedCourses` publication, to make sure it only returns a cursor with matching tagged courses.